drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/codec/ByteBufTransformer.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2013 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.drift.transport.netty.codec;
+
+import io.netty.buffer.ByteBuf;
+
+interface ByteBufTransformer
+{
+    ByteBuf transform(ByteBuf data);
+
+    ByteBuf untransform(ByteBuf data);
```

Comment:
what does transform and untransform mean? is this something related to Thrift?

Reply:
transform will transform the content of the payload that will be sending out (for example, encrypt it, compress it ... etc.).
untransform will do the opposite when we received a message.